### DISCUSSION TITLE
Removing PointLightBundle from Entity makes it disappear

### DISCUSSION THREAD

**Discussion by wbpaley** [bevyengine/bevy#11394] [Q&A] [answered]
Probably obvious to non-newbies--tia for your answer & tolerance!

I want objects to glow (be emissive) and shed light on surrounding objects (with a PointLightBundle) as I hover over them, then return to their former state when I stop hovering.

But when I try to remove the PointLightBundle (the very bottom of this code, my hacked standard bevy_mod_picking  minimal.rs) the whole object disappears. Why?

```
//! A minimal example of how to add picking support to your bevy app.

use bevy::prelude::*;
use bevy_mod_picking::debug::hide_pointer_text;
use bevy_mod_picking::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins.set(low_latency_window_plugin()))
        // All you need to do is add the picking plugin, with your backend of choice enabled in the
        // cargo features. By default, the bevy_mod_raycast backend is enabled via the
        // `backend_raycast` feature.
        .add_plugins(DefaultPickingPlugins)
        .add_systems(Startup, setup)

        .add_systems(Update, un_glow)

        .run();
}

fn setup(
    mut commands: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut materials: ResMut<Assets<StandardMaterial>>,
) {
    commands.spawn((
        PbrBundle {
            mesh: meshes.add(Mesh::from(shape::Plane::from_size(5.0))),
            material: materials.add(Color::rgb(0.3, 0.5, 0.3).into()),
            ..default()
        },
        PickableBundle::default(), // Optional: adds selection, highlighting, and helper components.
    ));
    commands.spawn(PointLightBundle {
        point_light: PointLight {
            intensity: 1500.0,
            shadows_enabled: true,
            ..default()
        },
        transform: Transform::from_xyz(4.0, 8.0, -4.0),
        ..default()
    });
    commands.spawn((Camera3dBundle {
        transform: Transform::from_xyz(3.0, 3.0, 3.0).looking_at(Vec3::ZERO, Vec3::Y),
        ..default()
    },));

    commands.spawn((
        PbrBundle {
            mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
            material: materials.add(StandardMaterial {
                base_color: Color::rgb(0.8, 0.7, 0.2).into(),
                emissive: Color::rgb(0.8, 0.7, 0.2).into(),
                ..default()
            }),
            transform: Transform::from_xyz(0.0, 0.5, 0.0),
            ..default()
        },
    ))

        //  put a light inside the cube
        .insert(PointLightBundle {
            point_light: PointLight {
                intensity: 5000.0,
                shadows_enabled: true,
                color: Color::rgb(0.8, 0.7, 0.2).into(),
                radius: 0.5,
                ..default()
            },
            transform: Transform::from_xyz(0.0, 0.5, 0.0),
            ..default()
        })

        //  send an event to remove the light after we hover & leave the cube
        .insert((
            PickableBundle::default(),
            On::<Pointer<Out>>::commands_mut(|event, commands| {
                commands.entity(event.target).insert(UnGlow(event.target));
            }),
        ));
}

#[derive(Component)]
pub struct UnGlow(Entity);

pub fn un_glow(
    mut query: Query<(&UnGlow, &mut Transform, &Handle<StandardMaterial>)>,
    mut commands: Commands
) {
    for (un_glow, transform, material_handle) in query.iter_mut() {

        //  expected to just remove the PointLightBundle
        //  instead, this makes the whole object disappear; todo: fix
        commands.entity(un_glow.0).remove::<PointLightBundle>();


        commands.entity(un_glow.0).remove::<UnGlow>();
    }
}
```

**Comment by alice-i-cecile** [comment]
By removing the whole bundle, you end up removing the critical [`SpatialBundle`](https://docs.rs/bevy/latest/bevy/render/prelude/struct.SpatialBundle.html) components on your entity, removing its visibility and position. Instead, you want to be more selective in terms of how you stop it from glowing.

**Reply by wbpaley** [reply]
Thanks! Will hunt for how to be more selective.

**Reply by wbpaley** [reply]
Didn't need to hunt; my first guess worked. Hope it's correct & idiomatic; do want to do things well here.

For other noobs who might need it to be spelled out, I just removed ```PointLight``` instead of ```PointLightBundle```. Do that in the above code and it works.

FWIW, I was expecting ```PointLight``` to be _inside_ ```PointLightBundle```, inside the ```Entity```--hierarchical--and I didn't know how to traverse the hierarchy to get to it. Guess Bevy/ECS magic finds it for me; still need to build my own mental model of how...

**Reply by wbpaley** [reply]
Tried to mark this as answer, too--it was, taught me, and worked.

**Comment by rparrett** [comment, marked as answer]
Most of the time, `PointLightBundle` and `PbrBundle` and other `Bundle`s aren't meant to be inserted onto the same entity. (A notable exception is `PickableBundle`, which you're using properly.

`PointLightBundle` and `PbrBundle` both share a subset of components: `Transform`, `GlobalTransform`, `Visibility`, `InheritedVisibility`, and `ViewVisibility` which are required for them to do their things.

When you remove `PointLightBundle`, you are removing those required components.

It's a bit of a fluke that `insert(PointLightBundle)` on an existing `PbrBundle` is doing what you want. This won't be the case for most built-in Bevy bundles. You should spawn a separate `PointLightBundle` entity as a child of your `PbrBundle` entity. You can then remove that child in `un_glow`. Check out the [hierarchy](https://github.com/bevyengine/bevy/blob/v0.12.1/examples/ecs/hierarchy.rs) example.

**Reply by wbpaley** [reply]
Lovely; thanks!

Do prefer understanding to magic...
